src/expr/src/scalar/func.rs

```diff
@@ -833,12 +833,22 @@ fn add_int64<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
         .map(Datum::from)
 }
 
-fn add_float32<'a>(a: Datum<'a>, b: Datum<'a>) -> Datum<'a> {
-    Datum::from(a.unwrap_float32() + b.unwrap_float32())
+fn add_float32<'a>(a: Datum<'a>, b: Datum<'a>) -> Result<Datum<'a>, EvalError> {
+    let sum = a.unwrap_float32() + b.unwrap_float32();
```

Comment:
I came across https://github.com/MaterializeInc/rust-dec but in its current state seemed cumbersome to use here.

- The explicit note that `Decimal32` intentionally does not support arithmetic gave me pause
- Using `Decimal64` involves more byte-twiddling than seemed necessary (though I'd be happy to modify the dec crate itself):
```
+    let mut c = Context::<Decimal64>::default();
+    let au = Decimal64::from_ne_bytes(a.unwrap_float64().to_ne_bytes());
+    let bu = Decimal64::from_ne_bytes(b.unwrap_float64().to_ne_bytes());
+    let sum = c.add(au, bu);
+    return if c.status().overflow() {
+        Err(EvalError::FloatOverflow)
+    } else {
+        Ok(Datum::from(f64::from_ne_bytes(sum.to_ne_bytes())))
+    };
```
- `Decimal32` and `Decimal64` exist so I wasn't sure if there was some reason why I shouldn't use `Decimal<N>` with `f32` / `f64`.

I figured I'd check in before modifying the `dec` crate -- especially since I already had this written up.  Happy to whip up a quick PR to make changes there if it does in fact make the most sense

Reply:
Ah, the decimal crate is only used for the `numeric` type. Is totally orthogonal to anything regarding float except for in as much as you can convert between them.